Hookies/Game/System/PowerupSystem.swift

```diff
@@ -199,10 +229,76 @@ class PowerupSystem: System, PowerupSystemProtocol {
     }
 
     private func isProtected(spriteComponent: SpriteComponent) -> Bool {
-        guard let powerup = powerups[spriteComponent] else {
+        guard let powerup = ownedPowerups[spriteComponent] else {
             return false
         }
         let hasShieldEffect = powerup.parent.get(ShieldEffectComponent.self) != nil
         return powerup.isActivated && hasShieldEffect
     }
 }
+
+
+// MARK: - Networking
+
+extension PowerupSystem {
+    private func registerNotificationObservers() {
```

Comment:
No caller for this function in the system. Can add it at the init method

Reply:
Done!